src/app/Library/CrudPanel/Traits/Read.php

```diff
@@ -82,8 +84,29 @@ public function autoEagerLoadRelationshipColumns()
     {
         $relationships = $this->getColumnsRelationships();
 
-        if (count($relationships)) {
-            $this->with($relationships);
+        foreach ($relationships as $relation) {
+            if (strpos($relation, '.') !== false) {
+                $relation_parts = explode('.', $relation);
+                $last_relation_part = array_pop($relation_parts);
+
+                $last_valid_relation_method = array_reduce(array_splice($relation_parts, 0, count($relation_parts)), function ($obj, $method) {
```

Comment:
@pxpm can you explain me this bit of code? 😅
Because, `array_splice($relation_parts, 0, count($relation_parts)` gives you the same thing, `$relation_parts,`right?
Also, I can't understand the logic behind `array_reduce` here.

Reply:
The intent of `array_reduce` is to reduce an array to a single value, but in this case, you always return a value, either the try catch fails or not. So I think it will always return a value relatively to the last element, independently of the previous results.
Let me know.

Reply:
basically what it does is that it starts from current model, check if the first element in array is a method in that model, if it is, we return the `related model` for the next interaction. We then proceed to check if those elements are subsquent relations.

I'v did a little debug script for you to have a better understanding:

For a `category.name` input as column name.

```php
array_reduce(array_splice($relation_parts, 0, count($relation_parts)), function ($obj, $method) use (&$last_valid_relation_method) {
                    Log::info('method is: ' . $method);
                    try {

                        Log::info('check if ' . $method . ' is a method in: ' . get_class($obj));
                        $result = $obj->$method();
                        $last_valid_relation_method = $method;
                        Log::info('it is. going to iterate again.');
                        return $result->getRelated();
                    } catch (Exception $e) {
                        Log::info($method . ' is not a method in: ' . get_class($obj));
                        return;
                    }
                }, $this->model);
```

```php
[2021-02-25 10:19:06] local.INFO: method is: category  
[2021-02-25 10:19:06] local.INFO: check if category is a method in: App\Models\Monster  
[2021-02-25 10:19:06] local.INFO: it is. going to iterate again.  
[2021-02-25 10:19:06] local.INFO: method is: name  
[2021-02-25 10:19:06] local.INFO: check if name is a method in: Backpack\NewsCRUD\app\Models\Category  
[2021-02-25 10:19:06] local.INFO: name is not a method in: Backpack\NewsCRUD\app\Models\Category 
```